### PR TITLE
Improved Devops Roadmap.

### PR DESCRIPTION
**Pull request by offensive-vk**
>[!NOTE]
> ## Description

- Add New Content for Some Topics
- Add New or Updated Links
- Add Proper Indent and EOF.
- Change Link Anchor(s) to Official.
- Add Missing Links Section
- Add Content to Topics with only Links.
- Enhanced Overall [Devops](https://roadmap.sh/devops) Roadmap Content.

---
*All Commits are signed using My GPG Key.*
**Thank you.**

### CODE REVIEW

**Comment by offensive-vk** [comment]
## Roadmap.sh on 1st now. ⚡
image

@kamranahmedse , please have a look at this pr.

File: src/data/roadmaps/devops/content/bash@syBIAL1mHbJLnTBoSxXI7.md [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
+
+Bash scripts usually have a .sh extension and are executable files. To run a Bash script, you need to give it execute permissions using the chmod command, and then you can run it by typing `./script_name.sh` in the terminal.
+
+## Shell Scripts are literally everywhere in DevOps
+
+- Automating system backups
+- Creating and managing user accounts
+- Configuring system settings
+- Running complex commands with a single command
+- Automating tasks that need to be performed on a regular basis
```
Comment (dansholds) [inline src/data/roadmaps/devops/content/bash@syBIAL1mHbJLnTBoSxXI7.md:15]:
```suggestion
```

No need for this part

File: src/data/roadmaps/devops/content/dns@Vu955vdsYerCG8G6suqml.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # DNS
 
-DNS (**D**omain **N**ame **S**ystem) is the phonebook of the Internet. Humans access information online through domain names, like nytimes.com or espn.com. Web browsers interact through Internet Protocol (IP) addresses. DNS translates domain names to IP addresses so browsers can load Internet resources.
+DNS (**D**omain **N**ame **S**ystem) is the phonebook of the Internet. Humans access information online through domain names, like `nytimes.com` or `espn.com.` Web browsers interact through Internet Protocol (IP) addresses. DNS translates domain names to IP addresses so browsers can load Internet resources.
```
Comment (dansholds) [inline src/data/roadmaps/devops/content/dns@Vu955vdsYerCG8G6suqml.md:3]:
```suggestion
DNS (Domain Name System) is the phonebook of the Internet. Humans access information online through domain names, like `nytimes.com` or `espn.com.` Web browsers interact through Internet Protocol (IP) addresses. DNS translates domain names to IP addresses so browsers can load Internet resources.
```

File: src/data/roadmaps/devops/content/networking-tools@OaqKLZe-XnngcDhDzCtRt.md [hunk 3]
```diff
@@ -63,5 +63,7 @@ IPtables is a command-line firewall utility that uses policy chains to allow or
 
 `SCP` is an acronym for Secure Copy Protocol.It is a command line utility that allows the user to securely copy files and directories between two locations usually between unix or linux systems.The protocol ensures the transmission of files is encrypted to prevent anyone with suspicious intentions from getting sensitive information.`SCP` uses encryption over an `SSH` (Secure Shell) connection, this ensures that the data being transferred is protected from suspicious attacks.
 
+Here are some of the resources to learn more:
```
Comment (dansholds) [inline src/data/roadmaps/devops/content/networking-tools@OaqKLZe-XnngcDhDzCtRt.md:66]:
```suggestion
Learn more from the following resources:
```

File: src/data/roadmaps/devops/content/performance-monitoring@gIEQDgKOsoEnSv8mpEzGH.md [hunk 4]
```diff
@@ -23,6 +23,8 @@ Short for **S**ystem **A**ctivity **R**eport, `sar` is a command line tool for U
 
 Short for **V**irtual **m**emory **stat**istic reporter, `vmstat` is a command line tool for Unix and Unix-like operating systems that reports various information about the operating system such as memory, paging, processes, I/O, CPU and disk usage.
 
+Here are some of the resources to learn more:
```
Comment (dansholds) [inline src/data/roadmaps/devops/content/performance-monitoring@gIEQDgKOsoEnSv8mpEzGH.md:26]:
```suggestion
Learn more from the following resources:
```

File: src/data/roadmaps/devops/content/what-is-and-how-to-setup-x@jCWrnQNgjHKyhzd9dwOHz.md [hunk 5]
```diff
@@ -2,9 +2,9 @@
...
-- Firewall
-- Caching Server
-- Web Server
\ No newline at end of file
+- [Forward Proxy](https://www.baeldung.com/nginx-forward-proxy)
+- [Reverse Proxy](https://www.ibm.com/docs/en/aspera-fasp-proxy/1.4?topic=proxy-configuring-server-forward)
+- [Load Balancer](https://cloud.google.com/load-balancing/docs/https/ext-https-lb-simple)
+- [Firewall](https://www.fortinet.com/resources/cyberglossary/firewall-configuration)
+- [Caching Server](https://www.quora.com/How-can-I-build-a-cache-server-at-home)
+- [Web Server](https://www.instructables.com/Set-up-your-very-own-Web-server/)
```
Comment (dansholds) [inline src/data/roadmaps/devops/content/what-is-and-how-to-setup-x@jCWrnQNgjHKyhzd9dwOHz.md:10]: Adding links like this in this topic doesn't make too much sense, the whole topic itself doesn't mean too much sense so it needs rethinking.

Just remove these additions for now.

**Review by dansholds — CHANGES_REQUESTED** [review]

**Comment by dansholds** [comment]
> ## Roadmap.sh on 1st now. ⚡
> 
> image
> 
> @kamranahmedse , please have a look at this pr.

I have reviewed, also please refrain from pinging maintainers, all PR _will_ be actioned.

**Comment by offensive-vk** [comment]
Aight mate

File: src/data/roadmaps/devops/content/datadog@BHny2Emf96suhAlltiEro.md [hunk 6]
```diff
@@ -1 +0,0 @@
-# Datadog
```
[the remark this answers is not in the document]
Comment (offensive-vk) [reply src/data/roadmaps/devops/content/datadog@BHny2Emf96suhAlltiEro.md:1]: There were three files with matching names. I deleted one with no content.

File: src/data/roadmaps/devops/content/what-is-and-how-to-setup-x@jCWrnQNgjHKyhzd9dwOHz.md (hunk 5 above)
Comment (offensive-vk) [reply src/data/roadmaps/devops/content/what-is-and-how-to-setup-x@jCWrnQNgjHKyhzd9dwOHz.md:10]: Sure

File: src/data/roadmaps/devops/content/datadog@BHny2Emf96suhAlltiEro.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (offensive-vk) [reply src/data/roadmaps/devops/content/datadog@BHny2Emf96suhAlltiEro.md:1]: Here's the path mate: 
https://github.com/kamranahmedse/developer-roadmap/blob/master/src/data/roadmaps/devops/content/datadog%40BHny2Emf96suhAlltiEro.md

[the remark this answers is not in the document]
Comment (dansholds) [reply src/data/roadmaps/devops/content/datadog@BHny2Emf96suhAlltiEro.md:1]: Thanks bruv

**Review by dansholds — APPROVED** [review]

**Pull request merged**